airflow/models/dag.py

```diff
@@ -1694,6 +1683,14 @@ def set_task_instance_state(
             include_upstream=False,
         )
 
+        if execution_date is None:
+            dag_run = (
+                session.query(DagRun).filter(DagRun.run_id == run_id, DagRun.dag_id == self.dag_id).one()
+            )  # Raises an error if not found
+            resolve_execution_date = dag_run.execution_date
+        else:
+            resolve_execution_date = execution_date
```

Comment:
Just to be sure, this is only refactoring so this is not unnecessarily called, and not related to the fix, right?

Reply:
Correct, yes.